lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -791,13 +823,16 @@ class TypeRefinementContextBuilder : private ASTWalker {
     OtherPlatformAvailabilitySpec *FoundOtherSpec = nullptr;
     PlatformVersionConstraintAvailabilitySpec *BestSpec = nullptr;
 
-    for (auto *Spec : available->getQueries()) {
+    auto Queries = available->getQueries();
+
+    for (auto *Spec : Queries) {
```

Comment:
It doesn't look like `Queries` is getting used later.
```suggestion
    for (auto *Spec : available->getQueries()) {
```